### ISSUE TITLE
Allow to run on a single core

### ISSUE THREAD

**Issue by endel** [gamestdio/colyseus#92]
- Versions < 0.4.x of the framework used to support **only** single-core, which is bad for production.
- Version 0.5.x **only** supports multi-core - which doesn't play nice with the majority of free hosting services.

By supporting single-core again, it would be easier to host the server for free on Heroku / Zeit / etc for prototyping without investing on servers.

**Comment by Federkun** [comment]
You should be careful with that. I already imagine someone that develop its application without having the cluster module in mind, and then he decide to enable it, expecting that should work as expected. So, I don't think that having something like

```
const gameServer = new Server({ server: httpServer, spawnWorkers: true/false });
```

is a good idea, since it's very likely that it will break until the flag is switched.

Right now I feel like colyseus it's trying to hide the fact that, under the hood, spawn multiple workers. But you can't really hide that complexity from the user, since if he need to know that if he want to build something with that.

**Comment by endel** [comment]
You're totally right @Federkun, thanks for your input!

**Comment by endel** [comment]
**UPDATE**:

- Previously, by using the `Server` class, the `ClusterServer` would be used under the hood, and child/fork processes would be created automatically.

- With the coming changes, `Server` runs everything on a single core - as it used to be on version `0.4.x` and below. (no breaking changes at this point)

- The `ClusterServer` now requires you to explicitly call `.fork()`, having an optional argument for the number of workers to spawn. That's a breaking change if you used `ClusterServer` by yourself.

https://github.com/gamestdio/colyseus/blob/98d066fb076f345b2dc90f87cf3226b3cf15a1d8/usage/ClusteredServer.ts#L8-L27

If you'd like to play around, the latest version is `colyseus@0.5.6-rc1`. I'll make it to `0.5.6` soon.

Do you like this approach, @Federkun?

Cheers!

**Comment by Federkun** [comment]
Hi @endel, I think that's the best way to provide multi-core support: good job. Right now I don't have much time available (I just moved to London) to test with it, but I'll try it once my situation here is more stable.

Thanks for your work

**Comment by endel** [comment]
Thanks @Federkun, I'll take semver more seriously for the next upcoming features. Just released version `0.5.6`.

Hope everything goes well for you in London!
